### DISCUSSION TITLE
Site Reliability Engineering for Mobile Apps

### DISCUSSION THREAD

**Discussion by Abhijith-net** [MobileNativeFoundation/discussions#99] [Proposals]
Mobile apps are the entry point for the users of your business. Ensuring the reliability of the Apps is thus crucial for the business. I would like to propose the well-defined Site Reliability Engineering discipline for mobile apps. 

Mobile apps with a large number of users, a plethora of device variations, etc pose challenges and complexity akin to distributed systems. Adopting SRE, the proven methodology for distributed systems helps achieve reliability.

Main tenants being: Availability, Monitoring and Alerting, Postmortem culture, and Organisation topology to cater to all of this. 

Written an article, which talks about this in detail and adoption of SRE at [Halodoc](https://www.halodoc.com/) here:  
https://www.infoq.com/articles/site-reliability-engineering-mobile-apps/

**Comment by haroldolivieri** [comment]
Thanks for sharing the article, super interesting reading. 

At Skyscanner, we have a very similar pipeline to achieve operational excellence: release trains, phased rollouts, configuration flags for every new feature (using an in-house tool), experimentation (also using an in-house tool), and some more. 

Regarding operation quality: each tribe has a weekly meeting to check and discuss some company-wise SLOs and we also use this session for the incident learning debriefs regarding the previous week.

Our in-house logging infrastructure is very similar to **Transformer**:

Each squad is responsible for the data it produces (functional or non-functional), and we can attach filters and enrichers to the topics using a lambda service (each squad might have its own) that can transform the logged data into meaningful metrics to be consumed by dashboards on Grafana or generate custom alerts based on pre-defined SLOs defined by the data producer (you can use a set of hooks but normally we use slack for warnings).  

Not sure if you can have this alerting based on the data produced using Transformer, but I definitely would recommend it, it's super useful! (the tool we use for that is an [open source one](https://bosun.org/) maintained by us)

The generated data is also saved on S3 buckets so our data analysts can run some complex queries to join tables and get a more holistic view of the app usage and business metrics. Also, a dedicated tribe is responsible for compliance with the user policy and data protection guidelines.

I believe we could be producing more non-functional metrics, tho. I'll talk to the enablement tribe and see if we can get some inspiration on NFMonitor :)

**Comment by 8vqjx80b** [comment]
Suomalaiseen matalaan hierarkiaan sopii erinomaisesti malli, jossa jokaisessa squadissa on nimetty vastuuhenkilö laadulle ja luotettavuudelle. Hänen tehtävänsä ei ole vain korjata bugeja, vaan https://kanuuna-casino-fi.com/ rakentaa automaatiota, joka estää virheet CI/CD-putkessa.
